README.md

```diff
@@ -1,109 +1,244 @@
+embarked      0.05    0.01  0.00 0.04   0.00   0.00    0.00  0.03      1.00
 ```
 
-Then, go to `http://localhost:8892`, browse the `nbs` folder, and run the notebooks in the specified order.
+The `ccc` function also has a `n_jobs` parameter that allows to control the number of CPU cores used.
+Below we compute the pairwise correlation between 20 features across 1000 objects:
 
-## Using Docker
+```python
+In [40]: data = np.random.rand(20, 1000)
 
-You can also run all the steps below using a Docker image instead of a local installation.
+In [41]: %timeit ccc(data, n_jobs=1)
+1.32 s ± 45.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
 
-```bash
-docker pull miltondp/clustermatch_gene_expr
+In [42]: %timeit ccc(data, n_jobs=2)
+771 ms ± 11 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
 ```
 
-The image only contains the conda environment with the code in this repository so, after pulling the image, you need to download the data as well:
 
-```bash
-mkdir -p /tmp/clustermatch_gene_expr_data
+## Reproducible research
 
-docker run --rm \
-  -v "/tmp/clustermatch_gene_expr_data:/opt/data" \
-  --user "$(id -u):$(id -g)" \
-  miltondp/clustermatch_gene_expr \
-  /bin/bash -c "python environment/scripts/setup_data.py"
-```
+Below we provide the steps to reproduce all the analyses in the CCC manuscript.
+
+### Setup
+
+To prepare the environment to run the analyses in the manuscript, follow the steps in [environment](environment/).
+After completing those steps, you'll have the source code in this repository, a Python environment (either using a Docker image or creating your own conda environment) and the necessary data to run the analyses.
 
-The `-v` parameter allows specifying a local directory (`/tmp/clustermatch_gene_expr_data`) where the data will be downloaded.
-If you want to generate the figures and tables for the manuscript, you need to clone the [manuscript repo](https://github.com/greenelab/clustermatch-gene-expr-manuscript) and pass it with `-v [PATH_TO_MANUSCRIPT_REPO]:/opt/manuscript`.
-If you want to change any other setting, you can set environmental variables when running the container; for example, to change the number of cores used to 2: `-e CM_N_JOBS=2`.
+### Running code
 
-You can run notebooks from the command line, for example:
+All the analyses are written as Jupyter notebooks and stored in the folder `nbs/`.
+All notebooks are organized by directories, such as `01_preprocessing`, with file names that indicate the order in which they should be run (if they share the prefix, then it means they can be run in parallel).
+You can run the analyses either using the JupyterLab server and your browser, or from the command line using [papermill](https://papermill.readthedocs.io/en/latest/).
+
+**Using the browser.** For example, let's say you want to run the preprocessing notebooks.
+If you want to use your browser, you first need to start the JupyterLab server:
```

Comment:
Maybe a personal choice, but it would be helpful to have the steps listed out in an outline for.
1. Setup server by calling `..`
2. Go to url ...
3. Browse to `nbs/`

Reply:
Perfect, I made several changes and now it's more clear. Thank you!